File: iosApp/Modules/Mocks/Use Cases/MockWalletNFTTracksUseCase.swift
import Foundation
import shared

class MockWalletNFTTracksUseCase: WalletNFTTracksUseCase {
	var walletSynced: any Kotlinx_coroutines_coreFlow { fatalError() }
	
	func getNFTTrack(id: String) -> NFTTrack? {
		nil
	}

	func getAllTracks() async throws -> [NFTTrack] {
		[]
	}
		
	func getAllTracksFlow() async throws -> any Kotlinx_coroutines_coreFlow {
		fatalError()
	}
	
	func getAllCollectableTracks() async throws -> [NFTTrack] {
		[]
	}
	
	func getAllCollectableTracksFlow() -> any Kotlinx_coroutines_coreFlow {
		fatalError()
	}
	
	func getAllNFTTracks() async throws -> [NFTTrack] {
		NFTTrack.mocks
	}
	
	func getAllNFTTracksFlow() -> any Kotlinx_coroutines_coreFlow {
		fatalError()
	}
	
	func getAllStreamTokens() async throws -> [NFTTrack] {
		NFTTrack.mocks
	}
	
	func getAllStreamTokensFlow() -> any Kotlinx_coroutines_coreFlow {
		fatalError()
	}
	
	func refresh() async throws {
		// no-op
	}
}

File: shared/src/commonMain/kotlin/io.newm.shared/commonPublic/usecases/mocks/MockWalletNFTTracksUseCase.kt
package io.newm.shared.commonPublic.usecases.mocks

import io.newm.shared.commonPublic.models.NFTTrack
import io.newm.shared.commonPublic.models.mocks.mockTracks
import io.newm.shared.commonPublic.usecases.WalletNFTTracksUseCase
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.flow

class MockWalletNFTTracksUseCase : WalletNFTTracksUseCase {
    override suspend fun refresh() {
        // no-op
    }

    private var _walletSynced = MutableStateFlow(false)

    override val walletSynced: Flow<Boolean>
        get() = _walletSynced.asStateFlow()

    override fun getAllCollectableTracksFlow(): Flow<List<NFTTrack>> = flow { emit(mockTracks) }

    override suspend fun getAllCollectableTracks(): List<NFTTrack> = mockTracks

    override fun getAllStreamTokensFlow(): Flow<List<NFTTrack>> = flow { emit(mockTracks) }

    override suspend fun getAllStreamTokens(): List<NFTTrack> = mockTracks

    override suspend fun getAllTracksFlow(): Flow<List<NFTTrack>> {
        TODO("Not yet implemented")
    }

    override suspend fun getAllTracks(): List<NFTTrack> {
        TODO("Not yet implemented")
    }
}
